multiqc/utils/report.py

```diff
@@ -363,35 +363,40 @@ def search_file(pattern, f, module_key):
     if pattern.get("contents") is not None or pattern.get("contents_re") is not None:
         if pattern.get("contents_re") is not None:
             repattern = re.compile(pattern["contents_re"])
-        try:
+        if not f.get("contents_lines") or config.filesearch_lines_limit < pattern.get("num_lines", 0):
+            f["contents_lines"] = []
             file_path = os.path.join(f["root"], f["fn"])
-            with io.open(file_path, "r", encoding="utf-8") as fh:
-                l = 1
-                for line in fh:
-                    # Search by file contents (string)
-                    if pattern.get("contents") is not None:
-                        if pattern["contents"] in line:
-                            contents_matched = True
-                            if pattern.get("fn") is None and pattern.get("fn_re") is None:
-                                return True
-                            break
-                    # Search by file contents (regex)
-                    elif pattern.get("contents_re") is not None:
-                        if re.search(repattern, line):
-                            contents_matched = True
-                            if pattern.get("fn") is None and pattern.get("fn_re") is None:
-                                return True
+            try:
+                with io.open(file_path, "r", encoding="utf-8") as fh:
+                    for i, line in enumerate(fh):
+                        f["contents_lines"].append(line)
+                        if i >= config.filesearch_lines_limit:
```

Comment:
I think that the above snippet is needed here, so that individual modules can override the global default in the maximum number of lines cached..?
```suggestion
                        if i >= config.filesearch_lines_limit and i >= pattern.get("num_lines", 0):
```

Reply:
Pushed this change, hope that's ok